### PR TITLE
Feature/k32w0/project structure

### PR DESCRIPTION
**Pull request by PetruSicoe**
K32W0 project structure refactoring
    
   Signed-off-by: PetruSicoe <petru-georgian.sicoe@nxp.com>

   **Problem**
   -  no place inside current project structure to add nxp platforms for example applications

**Change overview**

-    Place NXP K32W0 platform inside an nxp folder. 
   This prepares the ground for adding more platforms inside this folder
    
-    Code refactoring inside all the build files for
   k32w0 so that all the variable and file names
   refer to k32w0 specifically

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/platform/nxp/k32w/k32w0/GroupKeyStoreImpl.h [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        uint32_t StartTime; /**< The epoch key start time. */
+        uint32_t GlobalId;  /**< The application group key global ID. */
+    };
+};
+
+/**
+ * An implementation of the Chip GroupKeyStoreBase API for platforms based
+ * on the NXP SDK.
+ */
+class GroupKeyStoreImpl final : public ::chip::Profiles::Security::AppKeys::GroupKeyStoreBase, private K32WConfig
```
Comment (andy31415) [inline src/platform/nxp/k32w/k32w0/GroupKeyStoreImpl.h:55]: is this  supposed to be in this PR? Seems independent of a layout change

Comment (PetruSicoe) [reply src/platform/nxp/k32w/k32w0/GroupKeyStoreImpl.h]: Hi @andy31415 , thanks for the comment.
No, GroupKeyStoreImpl.h should not be in this PR, I see the file has been deleted in master by the time I pushed the commit.
Removed it from the project.

**Comment by PetruSicoe** [comment]
> ## - Break up this monolith and make it configurable.
> https://github.com/project-chip/connectedhomeip/blob/b521a983b8a0e05b0bfa9732b6113dbfaf0c80ca/third_party/k32w_sdk/nxp/k32w/k32w0/k32w0_sdk.gni#L251-L261
> 
> ###### This comment was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in [b521a98](https://github.com/project-chip/connectedhomeip/commit/b521a983b8a0e05b0bfa9732b6113dbfaf0c80ca) in #9425. cc @PetruSicoe.

Hi @mspang could you tell me what is exactly meant by monolith? Thank you!

**Comment by PetruSicoe** [comment]
Hi @andy31415 @mspang @bzbarsky-apple @Damian-Nordic @chrisdecenzo can you have a look on this PR and tell me if it's anything which still needs to be added in order to be merged?
Thank you!

**Review by Damian-Nordic — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**